source/write/replace.txt

```diff
@@ -0,0 +1,282 @@
+and sets its ``upsert`` field value to ``true``.  When ``opts`` is passed in as an optional argument, the ``replace_one()`` method
+searches the ``sample_restaurants.restaurants`` collection for a document with a ``name`` field value of ``In-N-Out Burger``. 
+Because this sample collection does not contain a document matching this query filter,
+the ``replace_one()`` method upserts a new document with a ``name`` field value of ``Shake Shack`` into the collection. 
+
+.. io-code-block::
+
+   .. input:: /includes/write/replace.cpp
+      :start-after: start-replace-options-upsert
+      :end-before: end-replace-options-upsert
+      :language: cpp
+      :dedent:
+
+   .. output:: 
+      :language: cli
+      :visible: false
+      
+      Total document count before replace_one(): 25359
+      Total document count after replace_one(): 25360
+
+Return Value
+~~~~~~~~~~~~
+
+The ``replace_one()`` method returns an instance of
+the ``mongocxx::result::replace`` class. This class contains the following
+member functions:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Field
+     - Description
+
+   * - ``matched_count()``
+     - | Returns the number of documents that matched the query filter, regardless of
+         how many were replaced.
+
+   * - ``modified_count()``
+     - | Returns number of documents modified by the replace operation. If a replaced
+         document is identical to the original, it is not included in this
+         count.
+         
+   * - ``result()``
+     - | Returns the bulk write result for the operation.
+
+   * - ``upserted_id()``
+     - | Returns the ID of the document that was upserted in the database, if the driver
+         performed an upsert.
+
+Example: ``matched_count()``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following example uses the ``replace_one()`` method to replace a document that has
+a ``name`` field value of ``Shake Shack`` with a new document that has a ``name`` 
+field value of ``In-N-Out Burger``. It then calls the ``matched_count()`` member 
+function to print the number of documents that match the search for a ``name`` field value of ``Shake Shack``:
+
+.. io-code-block::
+
+   .. input:: /includes/write/replace.cpp
+      :start-after: start-replace-result-matched
+      :end-before: end-replace-result-matched
+      :language: cpp
+      :dedent:
+
+   .. output:: 
+      :language: cli
+      :visible: false
+
+      Matched documents: 11
+
+Example: ``upserted_id()``
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following example uses the ``replace_one()`` method to find and replace a document that has a ``name`` field value of ``In-N-Out Burger``. 
+Because the upsert option is set to true, the {+driver-short+} inserts a new document when the query filter doesn't match any existing documents. 
+It then cals the ``upserted_id()`` member function to print the ``_id`` field value of the upserted document. 
```

Comment:
S: Some small formatting/wording suggestions:

```suggestion
The following example uses the ``replace_one()`` method to replace a document that has a ``name`` field value of ``"In-N-Out Burger"``. 
Because the ``upsert`` option is set to ``true``, the {+driver-short+} inserts a new document when the query filter doesn't match any existing documents. Then, the code
calls the ``upserted_id()`` member function to print the ``_id`` field value of the upserted document:
```